### PR TITLE
fix: Make the Details Page responsive

### LINKED ISSUE #14: Make the Details Page responsive
Presently the styles are disoriented and the arrow is shadowed by the title and the tags are invisible and also fix the text font sizes
image

We have to make it responsive for devices 
1. Greater than desktop
2. Desktop devices
3. Tablet devices
4. mobile devices

### PR DESCRIPTION
**Pull request by SankalpHaritash21**
## Summary
Adding Responsive Layout to details page

## Description
Adjusting Margin in detail-page. Now, adding responsiveness to details page according to Laptop, Tablet, Mobile.

## Images
Mobile:

Mo

Tablet:
Ta

Laptop:
La

## Issue(s) Addressed
- Closed #14 

## Review Checklist
Please make sure to go through this checklist before submitting your review.

- [X] Make sure the PR title is of format **`tag: Issue Title`**
   - tag should be one of `fix`,`feat`, `docs`, `chore`, `refactor`
   - Ex: `feat: Added responsive design for the home page`
- [X] Ensure the code follows our coding standards and guidelines
   - Ex:  naming conventions should be of form `file-name.ts` not `fileName.ts`
- [X] Ensure that you have staged only the required commits and raised the PR and not included all the files which would be otherwise required for local setup
- [X] Check if the PR is rebased to the latest main/development branch.
- [X] Make sure there are no unnecessary files or debugging artifacts in the PR.

### CODE REVIEW

**Comment by krishnaacharyaa** [comment]
Thank you @SankalpHaritash21 for the PR, have to admit there is significant improvement in your PR :)

**Comment by krishnaacharyaa** [comment]
image

Can we do this , push the content down 
And for tablet keep it 32px and also image to text i.e the detailed description which is there after top image please make sure that also has 32px gap, it is seeming more for me in the image.
And for mobile lets keep it 16px and 16px

**Comment by SankalpHaritash21** [comment]
Air should I make change in padding or in text size as 32 pzlx for tablet

**Comment by krishnaacharyaa** [comment]
> Sir should I make change in padding or in text size as 32px for tablet

Just the padding to 32px in 2 places in tablet and 16px in 2 places in mobile

**Comment by SankalpHaritash21** [comment]
sir should I give padding in left-right or top

**Comment by SankalpHaritash21** [comment]
Sir I have Updated the design.

Mobile:
Mobile

Tablet:
Tablet

**Comment by krishnaacharyaa** [comment]
Unfortunately you haven't got my point @SankalpHaritash21 



For your understanding I clearly explained via image and annotations, what is actually expected, let me know if you need any more info

**Comment by krishnaacharyaa** [comment]
> image
> 
> Can we do this , push the content down And for tablet keep it 32px and also image to text i.e the detailed description which is there after top image please make sure that also has 32px gap, it is seeming more for me in the image. And for mobile lets keep it 16px and 16px

Go through this again and Understand this and make changes accordingly

**Comment by krishnaacharyaa** [comment]
> sir should I give padding in left-right or top

For now , only top
To give you solution you need to use bottom-x value, rather than top-x in in classname. As it is absolute position.
Just make sure bottom-x is equivalent to bottom-[16px], pick x accordingly

**Comment by SankalpHaritash21** [comment]
Sir I have updated bottom-4 for mobile and bottom-8 for tablet

Mobile:
mobile


Tablet:
tab


code:

code

File: frontend/src/pages/details-page.tsx [hunk 1]
```diff
@@ -8,28 +8,32 @@ const DetailsPage = () => {
...
               </div>
             ))}
           </div>
-          <h1 className="text-4xl font-semibold ">{post.title}</h1>
+          <h1 className="text-lg md:text-2xl lg:text-4xl font-semibold ">{post.title}</h1>
         </div>
       </div>
-      <div className="px-64">
+      <div className="md:pl-32 lg:px-64  text-left w-full p-6">
         <div className="mt-4">
```
Comment (krishnaacharyaa) [inline frontend/src/pages/details-page.tsx:35]: padding we might have to maintain consistency here why we `pl` `px` and `p` is different cases
And you must have to add `pt` for description which needs to be same as the recent changes commented

**Comment by krishnaacharyaa** [comment]
> Sir I have updated bottom-4 for mobile and bottom-8 for tablet
> 
> Mobile: mobile
> 
> Tablet: tab
> 
> code:
> 
> code
> 
> Sir now it is correct or not

Looks good, but has enhancements, please dont send me the conde snippet push the commit and update the description I'll review from there.
What are the changes i am expecting ? 
image

Once you have made the required changes commit it re-request for review I'll review,
I hope this image should clear all of your doubts
Make sure you maintain consistency and improve existing styling if it weren't consistent

**Comment by SankalpHaritash21** [comment]
- Now I maintain code consistency.
- Improved existing  styling.
- Padding given according to the image.

File: frontend/src/pages/details-page.tsx (hunk 1 above)
Comment (SankalpHaritash21) [reply frontend/src/pages/details-page.tsx:35]: sir now I given padding px to maintain consistency and px can be use full

Comment (SankalpHaritash21) [reply frontend/src/pages/details-page.tsx:35]: px

Comment (SankalpHaritash21) [reply frontend/src/pages/details-page.tsx:35]: I think code is good to go @krishnaacharyaa  plz review

**Comment by SankalpHaritash21** [comment]
Sir can u tell me how can I request code review?

**Comment by krishnaacharyaa** [comment]
> Sir can u tell me how can I request code review?

image
Click on the circle like icon

**Comment by krishnaacharyaa** [comment]
> * Now I maintain code consistency.
> * Improved existing  styling.
> * Padding given according to the image.

Kindly update the description to have the updated image for confirmation

**Comment by SankalpHaritash21** [comment]
Updated image:

Mobile:
M

Laptop:



Tab:

Tab

File: frontend/src/pages/details-page.tsx (hunk 1 above)
Comment (krishnaacharyaa) [reply frontend/src/pages/details-page.tsx:35]: Yes looks good , thanks :)

File: frontend/src/pages/details-page.tsx [hunk 2]
```diff
@@ -8,28 +8,32 @@ const DetailsPage = () => {
...
               </div>
             ))}
           </div>
-          <h1 className="text-4xl font-semibold ">{post.title}</h1>
+          <h1 className="text-lg md:text-2xl lg:text-4xl font-semibold ">{post.title}</h1>
         </div>
       </div>
-      <div className="px-64">
+      <div className="px-4 md:px-8 lg:px-16 pt-4 md:pt-8 lg:pt-16  text-left w-full">
         <div className="mt-4">
```
Comment (krishnaacharyaa) [inline frontend/src/pages/details-page.tsx:35]: This mt-4 adds extra padding, please remove that or adjust accordingly. As per the requirement shown clearly in the image.

**Review by krishnaacharyaa — CHANGES_REQUESTED** [review]

**Comment by krishnaacharyaa** [comment]
> Updated image:
> 
> Mobile: M
> 
> Laptop:
> 
> L
> 
> Tab:
> 
> Tab

Please update in the original description, or it will become really difficult to track which image is posted when or i'll have to come everytime to the bottom of the page. Kindly read the instructions carefully from next time.

**Comment by SankalpHaritash21** [comment]
Sir should I need to change comment while commit changes

**Comment by krishnaacharyaa** [comment]
> Sir should I need to change comment while commit changes

image
This original comment yes, that is what i mean when i say update the description , it is also called PR description, the first comment of PR. (ideally the one which follows PR template is PR description)

**Comment by SankalpHaritash21** [comment]
Sir like this I have updated description

**Comment by krishnaacharyaa** [comment]
> Sir like this I have updated description

I dont see the changes :), in the PR description the alignment are not in order in the images, i am clueless what you are describing to

**Comment by SankalpHaritash21** [comment]
Sir I have removed margin from top and given gap between them. Sir I am not able to understand what changes I have to make

**Comment by krishnaacharyaa** [comment]
> Sir I have removed margin from top and given gap between them. Sir I am not able to understand what changes I have to make

image
Images in your PR description :)

**Comment by SankalpHaritash21** [comment]
Sir I have too update image also

**Comment by krishnaacharyaa** [comment]
All the images in the original PR description is outdated, you just have to update them with the latest images, which you have commented recently, just copy paste those from comment to original PR description

**Comment by krishnaacharyaa** [comment]
> Sir I have too update image also

Your PR description tells everything about the PR, so you have to update it whenever the reviewer asks you to change, that's the ideal way for long run , we can refer PR and understand instantly what are the changes done

**Comment by SankalpHaritash21** [comment]
sir done images updated in PR

File: frontend/src/pages/details-page.tsx (hunk 2 above)
Comment (SankalpHaritash21) [reply frontend/src/pages/details-page.tsx]: Margin Have been adjusted

**Review by krishnaacharyaa — APPROVED** [review]

**Comment by SankalpHaritash21** [comment]
sir I want to ask every time I have to update description and image for every pr

**Comment by krishnaacharyaa** [comment]
If the code changes includes UI changes then yes.
I dont know how you are taking it as, but to explain in simple words
you were adding new comments with the updated images which really makes this thread very difficult to work with as it keeps on increasing.
Instead you just override the images in the original description is what i meant.

**Comment by SankalpHaritash21** [comment]
ok sir I will keep it in mind.
I will update original PR description and image.
Sir every time I get 1 error by vercel whats that about

**Pull request merged**